### ISSUE TITLE
How to implement linear-control object for emissions intensity changes?

### ISSUE THREAD

**Issue by robbieorvis** [JGCRI/gcam-core#51]
Hello,

I'm wondering how to implement the linear-control emission factor setting as described here, http://jgcri.github.io/gcam-doc/emissions.html#user-options, to override the values GCAM will automatically calculate in response to GDP and population.

**Comment by ssmithClimate** [comment]
There are two parts to doing this.

First, you want to delete the existing GDP control object (or not read it in in the first place). Second, add the linear-control object:

Here is an example of doing both of these in one add-on XML file (although these can be done in separate add on XML files as well).

```
<?xml version="1.0" encoding="UTF-8"?>
<scenario>
   <world>
      <region name="India">
         <supplysector name="trn_freight_road" nocreate="1">
            <subsector name="Truck (5-9t)" nocreate="1">
               <stub-technology name="Liquids" nocreate="1">
                  <period year="1975">
                     <Non-CO2 name="BC">
                        <gdp-control delete="1" name="GDP_control"/>
                        <linear-control name="MFR">
                           <final-emissions-coefficient>7.63E-04</final-emissions-coefficient>
                           <start-year>2015</start-year>
                           <end-year>2030</end-year>
                        </linear-control>
                     </Non-CO2>
                  </period>
               </stub-technology>
            </subsector>
          </supplysector>
      </region>
   </world>
</scenario>
```
This example linearly ramps the 2015 emission factor for this technology down to a small value in 2030. 

This same input can be added to a future period (e.g.  <period year="2020">) so that it only takes effect from that vintage onward.

Note that we have identified a bug with the linear-control code in the current release. In some instances adding a linear-control will cause the code to error. We have a fix for this that will be in the next release, if you have problems let us know and we can help out with that.

**Comment by robbieorvis** [comment]
Steve, thanks for the guidance. This is exactly what I was looking for. Will wait to give this a shot until the bug fix is pushed.

**Comment by pralitp** [comment]
@robbieorvis, this bug is now fixed in GCAM 5.1.3
